app/models/course.rb

```diff
@@ -204,7 +193,7 @@ def new_draft_attributes
     end
   end
 
-  has_one :latest_published_enrichment, -> { published.order('created_at DESC, id DESC').limit(1) },
+  has_one :latest_published_enrichment, -> { published.most_recent },
```

Comment:
I did to compare and is the same as the one removed (but with the advantage of calling #preload or #includes or #eager_load)

```ruby
result = Courses::Query.new(params: {}).scope.select { |c| c.latest_published_enrichment != c.old_latest_published_enrichment }; puts result.size; puts result.first
0
=> nil
```